org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -16,58 +16,104 @@ namespace MixedReality.Toolkit.Input
     [AddComponentMenu("MRTK/Input/Near Interaction Mode Detector")]
     public class NearInteractionModeDetector : ProximityDetector
     {
+        /// <summary>
+        /// The set of near interactors that belongs to near interaction
+        /// </summary>
         [SerializeField]
         [Tooltip("The set of near interactors that belongs to near interaction")]
         private List<XRBaseInteractor> nearInteractors;
 
         /// <summary>
-        /// Used to keep track of the previously detected colliders so that we can know which
-        /// colliders stopped being detected and update their buttons front plate and rounded
-        /// rect if they are labeled as dynamic (based on proximity).
+        /// Keeps track of the previously detected interactables so that we can know which
+        /// interactable stopped being detected and trigger corresponding event.
+        /// </summary>
+        private readonly HashSet<IXRProximityInteractable> previouslyDetectedInteractables = new();
+
+        /// <summary>
+        /// Stores the currently detected interactables (updated every frame).
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private readonly List<IXRProximityInteractable> currentlyDetectedInteractables = new();
+
+        /// <summary>
+        /// Stores the interactables that are no longer detected (updated every frame).
+        /// </summary>
+        private readonly List<IXRProximityInteractable> noLongerDetectedInteractables = new();
 
         /// <inheritdoc />
         public override bool IsModeDetected()
```

Comment:
I still have some worry that this method is _not_ guaranteed to only be called once per-frame, but, in practice, each detector only has one controller assigned, so it _should_ only be once. We should keep an eye on that changing, though.

https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/95db47bc70e9d1f0393a19534460b2c682f5c0c4/org.mixedrealitytoolkit.input/InteractionModes/InteractionModeManager.cs#L412-L418

Reply:
This is also a `MonoBehaviour`. We _could_ run this code in `Update()` instead, to ensure once-per-frame.

Reply:
Thank you for the additional information.  We'll keep an eye on the controller-detector 1:1 relation changing.  In the meantime, resolving this conversation to be able to merge.